### PR TITLE
[13.0x.] ISPN-13844 ClassCastException with Oracle string-keyed-jdbc-store

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-13844

### CODE REVIEW

**Comment by wburns** [comment]
Updated to backport the test that reproduces the issue.

**Review by pruivo — APPROVED** [review]
LGTM

**Comment by pruivo** [comment]
merged! thanks @wburns !

File: integrationtests/server-integration/third-party-server/pom.xml [hunk 1]
```diff
@@ -18,14 +18,13 @@
         <version.arquillian-tomcat-managed>1.1.0.Final</version.arquillian-tomcat-managed>
         <ispnserver.project.dir>${basedir}/../../../server/runtime</ispnserver.project.dir>
         <ispn.config.file>${basedir}/../server-integration-commons/target/test-classes/infinispan-custom.xml</ispn.config.file>
-        <serverMemoryJvmArgs>-Xmx300m</serverMemoryJvmArgs>
```
Comment (diegolovison) [inline integrationtests/server-integration/third-party-server/pom.xml:21]: Those shouldn't be removed

File: integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/persistence/jdbc/util/BaseJdbcDeploy.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+public class BaseJdbcDeploy {
+
+    public static void addJdbcLibraries(WebArchive war) {
+        addLibrary(war, "org.infinispan:infinispan-cachestore-jdbc");
+        addLibrary(war, "org.infinispan:infinispan-core");
+        addLibrary(war, "com.h2database:h2");
+        addLibrary(war, "org.jgroups:jgroups");
+        war.addAsResource("connection.properties");
+        war.addAsLibrary(new File(System.getProperty("driver.path")));
```
Comment (diegolovison) [inline integrationtests/server-integration/third-party-server/src/test/java/org/infinispan/test/integration/thirdparty/persistence/jdbc/util/BaseJdbcDeploy.java:17]: Related failure: https://ci.infinispan.org/job/Infinispan/job/PR-10063/2/testReport/junit/org.infinispan.test.integration.thirdparty.persistence.jdbc.stringbased/PooledConnectionDeployIT/___/

**Pull request merged**